aspnetcore/blazor/tutorials/movie-database-app/part-6.md

```diff
@@ -0,0 +1,177 @@
+
+Change the `QuickGrid` component's <xref:Microsoft.AspNetCore.Components.QuickGrid.QuickGrid%601.Items%2A> parameter to use the `movies` collection:
+
+```diff
+- <QuickGrid Class="table" Items="DB.movies">
++ <QuickGrid Class="table" Items="@movies">
+```
+
+> [!NOTE]
+> Scaffolded code doesn't prefix component parameter values with `@` as a convention, but documentation examples always do. In the preceding example, assigning only `movie` without prefixing the `@` symbol (`Items="movies"`) is valid Razor syntax, but documentation examples always show such assignments with the `@` symbol (`Items="@movies"`) to indicate that C# is provided. When the `@` is explicitly added, the markup makes it clear that the value isn't a string literal being passed to `string`-typed parameter.
+
+The `s => s.Title.Contains()` code is a *lambda expression*. Lambdas are used in method-based LINQ queries as arguments to standard query operator methods such as the `Where` or `Contains` methods. LINQ queries aren't executed when they're defined or when they're modified by calling a method, such as `Where`, `Contains`, or `OrderBy`. Rather, query execution is deferred. The evaluation of an expression is delayed until its realized value is iterated.
+
+The <xref:System.Data.Objects.DataClasses.EntityCollection%601.Contains%2A> method is run on the database, not in the C# code. The case sensitivity of the query depends on the database and the collation. For SQL Server, `Contains` maps to [SQL `LIKE`](/sql/t-sql/language-elements/like-transact-sql), which is case insensitive. SQLite with default collation provides a mixture of case sensitive and case insensitive filtering, depending on the query. For information on making case insensitive SQLite queries, see the [Additional resources](#additional-resources) section of this article.
+
+Run the app and navigate to the movies `Index` page at `/movies`. The movies in the database load:
+
+![Mad Max movies before filtering in the movies Index page](~/blazor/tutorials/movie-database-app/part-6/_static/before-filtering.png)
+
+Append a query string to the URL in the address bar: `?titleSearch=road+warrior`. For example, the full URL appears as `https://localhost:7073/movies?titleSearch=road+warrior`, assuming the port number is `7073`. The filtered movie is displayed:
+
+!['The Road Warrior' Mad Max movie filtered using a query string in the browser's address bar](~/blazor/tutorials/movie-database-app/part-6/_static/query-string-filter-result.png)
+
+Next, give users a way to provide the `titleSearch` filter string via the component's UI. Add the following HTML under the H1 heading (`<h1>Index</h1>`). The following HTML reloads the page with the contents of the textbox as a query string value:
+
+```html
+<p>
+    <form action="/movies">
```

Comment:
I didn't think in this case with so few records that there would be a better user experience. Because this tutorial can't demo it, I wonder if a passing paragraph can tell readers what adding `data-enhance` does.

Reply:
Without `data-enhance`, when you submit the form you will get a full page reload. With `data-enhance`, Blazor will intercept the form post and update the DOM in place, so the page will update seamlessly and avoid requesting assets that were previously downloaded.